### PR TITLE
[XML Converter] Save Protobins within Godot 

### PR DESCRIPTION
**Pull request by klingbolt**
Replacing #159 

Changes to markers are saved in the split files. A save button takes those split files and saves them to individual files.

### CODE REVIEW

File: project.godot [hunk 1]
```diff
@@ -9,7 +9,7 @@
 config_version=4
 
 _global_script_classes=[ {
-"base": "Node",
+"base": "",
```
Comment (AsherGlick) [inline project.godot:12]: ?

File: Spatial.tscn [hunk 2]
```diff
@@ -784,6 +796,7 @@ margin_bottom = 304.0
 text = "Load Lutris Profile"
 
 [node name="CategoriesDialog" type="WindowDialog" parent="Control/Dialogs"]
+visible = true
```
Comment (AsherGlick) [inline Spatial.tscn:799]: ?

File: Spatial.tscn [hunk 3]
```diff
@@ -184,15 +195,16 @@ __meta__ = {
...
+margin_top = 89.0
+margin_right = 1448.0
+margin_bottom = 587.0
 window_title = "Open a Directory"
 mode = 2
 access = 2
-current_dir = ""
-current_path = ""
+current_dir = "."
+current_path = "."
```
Comment (AsherGlick) [inline Spatial.tscn:207]: ?

File: Spatial.tscn [hunk 4]
```diff
@@ -184,15 +195,16 @@ __meta__ = {
...
 [node name="ImportPackDialog" type="FileDialog" parent="Control/Dialogs"]
-margin_left = 289.0
-margin_top = 36.0
-margin_right = 960.0
-margin_bottom = 534.0
+visible = true
+margin_left = 777.0
+margin_top = 89.0
+margin_right = 1448.0
+margin_bottom = 587.0
```
Comment (AsherGlick) [inline Spatial.tscn:202]: ?

File: Spatial.gd [hunk 5]
```diff
@@ -632,31 +638,32 @@ func gen_new_icon(position: Vector3, texture_path: String, waypoint_icon, catego
...
+func save_current_map_data():
+	var packed_bytes = self.waypoint_data.to_bytes()
+	if packed_bytes.size() > 0:
+		var file = File.new()
+		file.open(self.marker_file_path, file.WRITE)
+		file.store_buffer(packed_bytes)
+
+func save_from_split_files():
+	save_current_map_data()
+	for waypoint_category in self.waypoint_data.get_category():
```
Comment (AsherGlick) [inline Spatial.gd:656]: Do we need to "save" every top-level category every time the button is pressed?

**Review by AsherGlick — CHANGES_REQUESTED** [review]
Can you split this up into 2 PRs.
* save to split proto from ingame state
* save to "export"? proto from split proto

From our previous conversation may have erroneously assumed that the former had already been completed and I think that there is going to be some discussion required on that point. While I have fewer comments about the latter.

**Comment by klingbolt** [comment]
This PR is now obsolete and its goals covered by other PRs

**Pull request closed without merging**